libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1146,6 +1345,27 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 			solUnimplemented("Fixed point types not implemented.");
 			break;
 		case Type::Category::Array:
+			if (_from == _to)
+			{
+				body = "converted := value";
+				break;
+			}
+			else
+			{
+				ArrayType const& from = dynamic_cast<decltype(from)>(_from);
+				ArrayType const& to   = dynamic_cast<decltype(to)>  (_to);
+
+				ReferenceType const& fromNoPtr =
```

Comment:
Maybe a helper `equalUpToPointer` in the array class would be nice.

Reply:
Oh actually you can just compare `from.mobileType() == to.mobileType()`.

Reply:
I just tried using `mobileType()` and the result is not the same (my tests fail and the assert here is reached)

Reply:
For which types does that happen?

Reply:
```
(gdb) print from.richIdentifier()
$6 = "t_array(t_uint256)dyn_memory"
(gdb) print to.richIdentifier()
$7 = "t_array(t_uint256)dyn_memory_ptr"
```

Reply:
hmm maybe I made a mistake and forget to dereference it.. I'll check again

Reply:
Okay, it was a missing derefernce. Works now! :+1:

Reply:
Nice!